src/schema/TransferSummary/Query.ts

```diff
@@ -8,68 +8,121 @@
  *       Yevhen Kyriukha <yevhen.kyriukha@modusbox.com>                   *
  **************************************************************************/
 
-import { arg, extendType, inputObjectType, intArg } from 'nexus';
-import { NexusGenObjects } from 'nexus-typegen';
+import { list, stringArg, extendType, intArg, inputObjectType } from 'nexus';
 
+// Define input type for TransferSummaryFilter
 const TransferSummaryFilter = inputObjectType({
   name: 'TransferSummaryFilter',
   definition(t) {
-    t.field('startDate', { type: 'DateTimeFlexible' });
-    t.field('endDate', { type: 'DateTimeFlexible' });
-    t.field('errorCode', { type: 'Int' });
-    t.field('payerDFSP', { type: 'String' });
-    t.field('payeeDFSP', { type: 'String' });
-    t.field('currency', { type: 'Currency' });
+    t.dateTimeFlex('startDate');
+    t.dateTimeFlex('endDate');
   },
 });
 
+// Create where condition for filtering
+const createWhereCondition = (filter: any) => {
+  const whereCondition: any = {};
+
+  if (filter?.startDate) {
+    whereCondition.createdAt = {
+      gte: new Date(filter.startDate),
+    };
+  }
+
+  if (filter?.endDate) {
+    whereCondition.createdAt = {
+      ...whereCondition.createdAt,
+      lte: new Date(filter.endDate),
+    };
+  }
+
+  return whereCondition;
+};
+
 const Query = extendType({
   type: 'Query',
   definition(t) {
-    t.nonNull.list.nonNull.field('transferSummary', {
+    // Define a field to fetch transfer summaries
+    t.nonNull.list.field('transferSummary', {
       type: 'TransferSummary',
       args: {
-        filter: arg({ type: 'TransferSummaryFilter' }),
         limit: intArg(),
         offset: intArg(),
+        filter: TransferSummaryFilter,
+        groupBy: list(stringArg()),
       },
-      resolve: async (parent, args, ctx, info) => {
-        const fields = ctx.getRequestFields(info) as NexusGenObjects['TransferSummary'];
+      resolve: async (parent, args, ctx) => {
+        const { limit = 100, offset = 0, filter = {}, groupBy = [] } = args;
+        const whereCondition = createWhereCondition(filter);
+
+        const groupByFields = groupBy && groupBy.length > 0 ? groupBy : null;
+
+        let aggregateResult;
+
+        try {
+          if (!groupByFields) {
+            // Aggregate results without grouping when no group by fields are provided
+            aggregateResult = await ctx.transaction.transaction.aggregate({
+              _count: { transferId: true },
+              _sum: { sourceAmount: true, targetAmount: true },
+              where: whereCondition,
+              skip: offset ?? 0,
+              take: limit ?? 100,
```

Comment:
We need to test and analyse the queries for getting these aggregations if we use indexes efficiently. Because these aggregations are used in pie charts which load every time in the tranfers UI for each operation.

Reply:
I can add indexing for these as they are not nested fields.